### PR TITLE
Rename libcudf++ to libcudf.

### PR DESCRIPTION
**Pull request by bdice**
## Description
For consistency across our documentation, this PR renames `libcudf++` to `libcudf`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by davidwendt — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Review by vyasr — APPROVED** [review]
LGTM. I committed one additional change to the line referencing the original libcudf++ presentation. If you think it's more accurate to continue calling that particular instance libcudf++ feel free to revert that change, making the change was the easiest way to point it out.

**Review by ttnghia — APPROVED** [review]
Great change :laughing:

File: cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ Resource ownership is an essential concept in libcudf. In short, an "owning" obj
 resource (such as device memory). It acquires that resource during construction and releases the
 resource in destruction ([RAII](https://en.cppreference.com/w/cpp/language/raii)). A "non-owning"
 object does not own resources. Any class in libcudf with the `*_view` suffix is non-owning. For more
-detail see the [`libcudf++` presentation.](https://docs.google.com/presentation/d/1zKzAtc1AWFKfMhiUlV5yRZxSiPLwsObxMlWRWz_f5hA/edit?usp=sharing)
+detail see the [`libcudf` presentation.](https://docs.google.com/presentation/d/1zKzAtc1AWFKfMhiUlV5yRZxSiPLwsObxMlWRWz_f5hA/edit?usp=sharing)
```
Comment (jrhemstad) [inline cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md:179]: To be fair, this one is the only one that could still stay as `libcudf++`. More importantly, these slides should probably be somewhere other than my Google Drive... or just remove the link all together. The content within this guide should supersede anything in those slides by this point.

Comment (bdice) [reply cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md:179]: I’m going to leave it as-is. The presentation is titled libcudf++ but it’s about libcudf and I don’t want readers to be confused about the proper name of the library.

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**